lib/carousel_slider.dart

```diff
@@ -215,7 +215,7 @@ class _CarouselSliderState extends State<CarouselSlider> with TickerProviderStat
       },
       itemBuilder: (BuildContext context, int i) {
         final int index =
-            _getRealIndex(i + widget.initialPage, widget.realPage, widget.items.length);
```

Comment:
Instead of removing `+ widget.initialPage` from line 218, try adding it to line 211.
We are calculating the same thing twice, but in two different ways.
Adding offset to both will return the same index, adjusted for offset, to both.